core/src/environment.rs

```diff
@@ -348,15 +296,20 @@ mod tests {
         let _ = env_base.clone();
         assert_eq!(env_base.iter_layers().count(), 1);
         env_base.insert(2, 'b');
+        assert_eq!(env_base.iter_layers().count(), 1);
+
+        let _env2 = env_base.clone();
+        env_base.insert(3, 'c');
         assert_eq!(env_base.iter_layers().count(), 2);
-        let _ = env_base.clone();
         let mut iter = env_base.iter_layers();
         let map1 = iter.next().unwrap();
-        assert_eq!(map1.get(&2), Some(&'b'));
+        assert_eq!(map1.get(&3), Some(&'c'));
+        assert_eq!(map1.get(&2), None);
         assert_eq!(map1.get(&1), None);
         let map2 = iter.next().unwrap();
+        assert_eq!(map2.get(&3), None);
+        assert_eq!(map2.get(&2), Some(&'b'));
         assert_eq!(map2.get(&1), Some(&'a'));
-        assert_eq!(map2.get(&2), None);
         assert!(iter.next().is_none());
```

Comment:
I find a bit counter intuitive that the next layer is empty. Where has `a` gone?

Reply:
I don't understand: `c` is in the top layer (`map1`) and `a` and `b` are in the previous layer `map2`.

Reply:
My bad, I had trouble reading the diff, but looking at the final version alone it makes sense. I thought `a` disappeared from the iterator.